### PR TITLE
[PhpUnitBridge] testing for deprecations is not risky

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Bridge/PhpUnit/Legacy/SymfonyTestsListenerTrait.php [hunk 1]
```diff
@@ -172,6 +173,10 @@ public function addSkippedTest($test, \Exception $e, $time)
     public function startTest($test)
     {
         if (-2 < $this->state && ($test instanceof \PHPUnit_Framework_TestCase || $test instanceof TestCase)) {
+            if (null !== $test->getTestResultObject()) {
```
Comment (xabbuh) [inline src/Symfony/Bridge/PhpUnit/Legacy/SymfonyTestsListenerTrait.php]: `getTestResultObject()` can return `null`, for example, when the test case leads to a PHP warning. In this case, `$test` is an instance of `PHPUnit\Framework\WarningTestCase` which does not emit a test result object.

**Review by nicolas-grekas — APPROVED** [review]
:+1:

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Comment by stof** [comment]
couldn't we ensure that checks for ``@expectedDeprecation`` are registered in the assertion count instead ? This would also allow getting the assertion count right.

**Comment by nicolas-grekas** [comment]
Good idea! Is that the case for `@expectedException`? @xabbuh could you look at it again please?

**Comment by xabbuh** [comment]
It's the case for `@expectedException`. But that's hardcoded in the base `TestCase` class. Not sure how we could hook into that.

**Comment by nicolas-grekas** [comment]
Trigger a dummy `$test->assertTrue(true)` to increase the counter?

**Comment by stof** [comment]
better: use ``$test->addToAssertionCount($count)``

**Comment by stof** [comment]
and this even allows to add the proper number of assertions (one per expected deprecation)

**Comment by xabbuh** [comment]
see #21828

**Pull request merged**